extensions/notebook/src/book/remoteBookController.ts

```diff
@@ -0,0 +1,290 @@
+		}
+		return await this._book.createLocalCopy();
+	}
+
+	public async fetchGithubReleases(url: URL): Promise<IReleases[]> {
+		let options = {
+			headers: {
+				'User-Agent': 'request'
+			}
+		};
+		return new Promise<IReleases[]>((resolve, reject) => {
+			request.get(url.href, options, (error, response, body) => {
+				if (error) {
+					return reject(error);
+				}
+
+				if (response.statusCode === 404) {
+					return reject(msgResourceNotFound);
+				}
+
+				if (response.statusCode !== 200) {
+					return reject(response.statusCode);
+				}
+				let releases = JSON.parse(body);
+				let bookReleases: IReleases[] = [];
+				if (releases) {
+					let keys = Object.keys(releases);
+					keys = keys.filter(key => {
+						let release = {} as IReleases;
+						try {
+							release.name = releases[key].name;
+							release.assets_url = new URL(releases[key].assets_url);
+						}
+						catch (error) {
+							return reject(error);
+						}
+						bookReleases.push(release);
+					});
+				}
+				if (bookReleases) {
+					this.model.releases = bookReleases;
+					resolve(bookReleases);
+				} else {
+					return reject(msgReleaseNotFound);
+				}
+			});
+		});
+	}
+
+	public async fecthListAssets(release: IReleases): Promise<IAssets[]> {
+		let format: string[] = [];
+		if (utils.getOSPlatform() === utils.Platform.Windows || utils.getOSPlatform() === utils.Platform.Mac) {
+			format = ['zip'];
+		} else {
+			format = ['tar.gz', 'tgz'];
+		}
+		let options = {
+			headers: {
+				'User-Agent': 'request'
+			}
+		};
+		return new Promise<IAssets[]>((resolve, reject) => {
+			request.get(release.assets_url.href, options, (error, response, body) => {
+				if (error) {
+					return reject(error);
+				}
+
+				if (response.statusCode === 404) {
+					return reject(msgResourceNotFound);
+				}
+
+				if (response.statusCode !== 200) {
+					return reject(response.statusCode);
+				}
+				let assets = JSON.parse(body);
+				let githubAssets: IAssets[] = [];
+				if (assets) {
+					let keys = Object.keys(assets);
+					keys = keys.filter(key => {
+						let asset = {} as IAssets;
+						asset.url = new URL(assets[key].url);
+						asset.name = assets[key].name;
+						asset.browser_download_url = new URL(assets[key].browser_download_url);
+						let groupsRe = asset.name.match(this.assetNameRE);
+						if (groupsRe !== null) {
+							asset.book = groupsRe[1];
+							asset.version = groupsRe[2];
+							asset.language = groupsRe[3];
+							asset.format = groupsRe[4];
+							if (format.includes(asset.format)) {
+								githubAssets.push(asset);
+							}
+						}
+					});
+				}
+				if (githubAssets.length > 0) {
+					this.model.assets = githubAssets;
+					resolve(githubAssets);
+				}
+				return reject(msgBookNotFound);
+			});
+		});
+	}
+
+	public getReleases(): IReleases[] {
+		return this.model.releases;
+	}
+
+	public getAssets(): IAssets[] {
+		return this.model.assets;
+	}
+}
+
+export interface IReleases {
```

Comment:
Just `IRelease` seems more accurate here - this is only going to be a single release